### PR TITLE
StaticAnalysis update for changing Signature verification severity issue

### PR DESCRIPTION
**Pull request by shahabhijeet**
This checklist is used to make sure that common issues in a pull request are covered by the creator. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).

Below in **Overall Changes**, check off the boxes that apply to your PR. Within each of the categories that you select, make sure that you can check off **all** of the boxes.
## Overall Changes
- [x] [**MANDATORY** - General changes](#general)
- [x] [**MANDATORY** - Add/remove/edit test(s)](#tests)

General Changes
This PR fixes the below two issues
[Issue# 2845](https://github.com/Azure/azure-powershell/issues/2610)

StaticAnaylysis changes
1) Added tests to now verify static analysis rules for SignatureVerification
2) You can now add cmdLets to the test project and the test can now target specific verification rather than running all the analysis verifier even though if they are not applicable.
3) Added log4net logger as the overall logger for the StaticAnalysis tool
4) Disabled (and not deleted) consoleLogger, as log4net can handle logging to console logger.
5) Currently the changes are more targeted towards SignatureVerification, but later on help as well as dependency verifier will be updated accordingly.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@shahabhijeet__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (abhishah). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by shahabhijeet** [comment]
Fix for Issue: https://github.com/Azure/azure-powershell/issues/2610

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1118/

**Comment by shahabhijeet** [comment]
Test this please

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1131/

**Comment by markcowl** [comment]
@shahabhijeet Please remove the system.management.automation changes from this PR.  Those are needed for this release and will be included in a targeted PR for the needed fix.

File: src/Common/Commands.Common/Commands.Common.csproj [hunk 1]
```diff
@@ -103,7 +103,10 @@
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
-    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
+    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (markcowl) [inline src/Common/Commands.Common/Commands.Common.csproj]: @shahabhijeet these should all be simple framework references, not file references.

File: tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            try
+            {
+                Log.LogMessage("Verifying Authenticode Signature for '{0}' - '{1}'", assemblyName, assemblyPath);
+                X509Certificate cert = X509Certificate.CreateFromSignedFile(assemblyPath);
+                if ((assemblyName.ToLower().StartsWith("microsoft")) || (assemblyName.ToLower().StartsWith("system")))
+                {
+                    if (cert.Issuer.ToLower().Contains("microsoft"))
+                    {
+                        // Verify if the certificate by which it was signed is not expired
+                        DateTime notAfter = DateTime.Parse(cert.GetExpirationDateString());
```
Comment (markcowl) [inline tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs]: I think we need to do a full valididty check here, including crl.  Are there simple APIs we can use to detect that the cert is valid.

File: tools/WindowsAzurePowerShell.Test.Commands/WindowsAzurePowerShell.Test.Commands/WindowsAzurePowerShell.Test.Commands.csproj [hunk 3]
```diff
@@ -38,7 +38,10 @@
       <HintPath>..\packages\WindowsAzure.Storage.2.1.0.3\lib\net40\Microsoft.WindowsAzure.Storage.dll</HintPath>
     </Reference>
     <Reference Include="System.Data.Services.Client" />
-    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
+    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (markcowl) [inline tools/WindowsAzurePowerShell.Test.Commands/WindowsAzurePowerShell.Test.Commands/WindowsAzurePowerShell.Test.Commands.csproj]: this should not be a file reference

**Comment by markcowl** [comment]
@shahabhijeet Please change back the PR title when you have made the changes we discussed

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj [hunk 4]
```diff
@@ -119,10 +119,7 @@
     </Reference>
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj]: rather than reverting them explicitly can't you just check out the previous versions of these files?

File: tools/StaticAnalysis/StaticAnalysis.Test/packages.config [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
+  <package id="System.Management.Automation" version="6.1.7601.17515" targetFramework="net45" />
```
Comment (markcowl) [inline tools/StaticAnalysis/StaticAnalysis.Test/packages.config]: ??

File: build.proj [hunk 6]
```diff
@@ -58,6 +58,12 @@
     <StaticAnalysis Include=".\tools\StaticAnalysis\StaticAnalysis.sln" />
   </ItemGroup>
   
+  <!-- Tasks -->
+  <UsingTask TaskName="ValidateStrongNameSignatureTask" AssemblyFile="$(LibraryToolsFolder)\Microsoft.Azure.Build.Tasks.dll"  />
+  <UsingTask TaskName="FilterOutAutoRestLibraries" AssemblyFile="$(LibraryToolsFolder)\Microsoft.Azure.Build.Tasks.dll"  />
+  <UsingTask TaskName="VerifyAuthenticodeSignatureTask" AssemblyFile="$(LibraryToolsFolder)\Microsoft.Azure.Build.Tasks.dll"  />
```
Comment (markcowl) [inline build.proj]: I thought you were removing the Authenticode task?

File: src/ResourceManager/Resources/Commands.Resources/Commands.Resources.csproj [hunk 7]
```diff
@@ -128,7 +128,10 @@
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
-    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
+        <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/Commands.Resources.csproj]: Revert this change

File: src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/Commands.ExpressRoute.csproj [hunk 8]
```diff
@@ -91,9 +91,9 @@
       <HintPath>..\..\..\packages\Microsoft.WindowsAzure.Management.ExpressRoute.0.19.0-preview\lib\net40\Microsoft.WindowsAzure.Management.ExpressRoute.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
+    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (markcowl) [inline src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/Commands.ExpressRoute.csproj]: This is the one that needs to change - if you checkout the latest, the change to a simple framework reference should be there

Comment (markcowl) [inline src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/Commands.ExpressRoute.csproj]: Revert changes to this file

File: tools/BuildPackagesTask/Microsoft.Azure.Build.Tasks.csproj [hunk 9]
```diff
@@ -35,18 +35,26 @@
...
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="DebugTask.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="FilterOutAutoRestLibraries.cs" />
     <Compile Include="RegexReplacementTask.cs" />
     <Compile Include="StrongNameUtility.cs" />
     <Compile Include="ValidateStrongNameSignatureTask.cs" />
+    <Compile Include="VerifyAuthenticodeSignatureTask.cs" />
```
Comment (markcowl) [inline tools/BuildPackagesTask/Microsoft.Azure.Build.Tasks.csproj]: Remove the authenticode changes, as discussed

File: tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
+﻿namespace Microsoft.Azure.Build.Tasks
```
Comment (markcowl) [inline tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs]: Remove, as discussed.

File: tools/BuildPackagesTask/Tests/AuthenticodeSignedTaskSmokeTest.proj [hunk 11]
```diff
@@ -0,0 +1,27 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (markcowl) [inline tools/BuildPackagesTask/Tests/AuthenticodeSignedTaskSmokeTest.proj]: Remove, as discussed.

**Comment by markcowl** [comment]
@shahabhijeet A couple of comments
